air-script/tests/pub_inputs/pub_inputs.masm

```diff
@@ -1,16 +1,70 @@
     padw mem_loadw.4294966016 movdn.3 movdn.3 drop drop ext2mul
 end # END PROC compute_evaluate_transitions
 
+proc.compute_evaluate_boundary
+    # boundary constraint 0 for main
+    padw mem_loadw.4294965000 movdn.3 movdn.3 drop drop
+    # Load public input stack_inputs pos 0
+    padw mem_loadw.4294960000 movdn.3 movdn.3 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966016 drop drop ext2mul
+    # boundary constraint 1 for main
+    padw mem_loadw.4294965001 movdn.3 movdn.3 drop drop
+    # Load public input stack_inputs pos 1
+    padw mem_loadw.4294960000 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966017 movdn.3 movdn.3 drop drop ext2mul
+    # boundary constraint 2 for main
+    padw mem_loadw.4294965002 movdn.3 movdn.3 drop drop
+    # Load public input stack_inputs pos 2
+    padw mem_loadw.4294960001 movdn.3 movdn.3 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966017 drop drop ext2mul
+    # boundary constraint 3 for main
+    padw mem_loadw.4294965003 movdn.3 movdn.3 drop drop
+    # Load public input stack_inputs pos 3
+    padw mem_loadw.4294960001 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966018 movdn.3 movdn.3 drop drop ext2mul
+    # boundary constraint 4 for main
+    padw mem_loadw.4294965000 movdn.3 movdn.3 drop drop
+    # Load public input stack_outputs pos 0
+    padw mem_loadw.4294960000 movdn.3 movdn.3 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966018 drop drop ext2mul
+    # boundary constraint 5 for main
+    padw mem_loadw.4294965001 movdn.3 movdn.3 drop drop
+    # Load public input stack_outputs pos 1
+    padw mem_loadw.4294960000 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966019 movdn.3 movdn.3 drop drop ext2mul
+    # boundary constraint 6 for main
+    padw mem_loadw.4294965002 movdn.3 movdn.3 drop drop
+    # Load public input stack_outputs pos 2
+    padw mem_loadw.4294960001 movdn.3 movdn.3 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966019 drop drop ext2mul
+    # boundary constraint 7 for main
+    padw mem_loadw.4294965003 movdn.3 movdn.3 drop drop
+    # Load public input stack_outputs pos 3
+    padw mem_loadw.4294960001 drop drop ext2sub
+    # Multiply by the composition coefficient
+    padw mem_loadw.4294966020 movdn.3 movdn.3 drop drop ext2mul
+end # END PROC compute_evaluate_boundary
+
 proc.evaluate_transitions
     exec.compute_evaluate_transitions
     # Accumulate the numerator of the constraint polynomial
     ext2add
 end # END PROC evaluate_transitions
 
 proc.evaluate_boundary
+    exec.compute_evaluate_boundary
+    # Accumulate the numerator of the constraint polynomial
+    ext2add ext2add ext2add ext2add ext2add ext2add ext2add ext2add
 end # END PROC evaluate_boundary
```

Comment:
minor nit: I'd probably add `constraints` suffix to procedure names here - i.e., `evaluate_integrity_constraints` and `evaluate_boundary_constraints`.

Reply:
renamed